File: js/programmers/level.0/prgm_181832_1.js
function solution(n) {
  let result = Array.from(Array(n), () => Array(n).fill(0));
  let row = 0;
  let col = 0;
  let num = 1;

  for (i = n; i > 0; i -= 2) {
    for (j = 0; j < i; j++) {
      result[row][col] = num;
      col++;
      num++;
    }
    col--;
    row++;

    for (j = 0; j < i - 1; j++) {
      result[row][col] = num;
      row++;
      num++;
    }
    col--;
    row--;

    for (j = 0; j < i - 1; j++) {
      result[row][col] = num;
      col--;
      num++;
    }
    col++;
    row--;

    for (j = 0; j < i - 2; j++) {
      result[row][col] = num;
      row--;
      num++;
    }
    col++;
    row++;
  }
  return result;
}

File: python/programmers/level.0/prgm_181832_1.py
def solution(n):
    result = [[0 for _ in range(n)] for _ in range(n)]
    row = 0
    col = 0
    num = 1

    for i in range(n, 0, -2):
        for _ in range(i):
            result[row][col] = num
            col += 1
            num += 1
        col -= 1
        row += 1

        for _ in range(i-1):
            result[row][col] = num
            row += 1
            num += 1
        col -= 1
        row -= 1

        for _ in range(i-1):
            result[row][col] = num
            col -= 1
            num += 1
        col += 1
        row -= 1

        for _ in range(i-2):
            result[row][col] = num
            row -= 1
            num += 1
        col += 1
        row += 1
    
    return result

